rockmoventstrategy.cpp

```diff
@@ -6,7 +6,74 @@ std::vector<BoardCellPosition> RockMoventStrategy::getAvailableMoves(const Chess
     std::vector<BoardCellPosition> result {};
     const GameBoard& board {GameBoard::instance()};
     const BoardCellPosition& currentPosition = piece.position();
+    size_t count;
 
+    //0 - place row
```

Comment:
Can we merge row and column algorithms into one?
Also, find_if is not used, as described in task.